dev/performance/perf2.py

```diff
@@ -0,0 +1,176 @@
+##############################################################################
```

Comment:
Merged with minor pep8/formatting changes.

Note, running the benchmark with the memory measurement turned on is a lot slower. That is why I measure speed and memory usage separately in the existing benchmark.